plutarch-test/Main.hs

```diff
@@ -8,24 +8,36 @@ import qualified ExtraSpec
 #if MIN_VERSION_GLASGOW_HASKELL(9,0,0,0)
 import qualified Plutarch.FieldSpec as FieldSpec
 import qualified Plutarch.MonadicSpec as MonadicSpec
+import Plutarch.Test.Run (runPlutarchSpec)
+import Test.Syd (Spec, describe)
 #else
 import qualified Plutarch.FFISpec as FFISpec
+import Test.Syd (Spec, describe, sydTest)
 #endif
 
 import GHC.IO.Encoding (setLocaleEncoding, utf8)
-import Test.Syd (describe, sydTest)
 
 main :: IO ()
 main = do
   setLocaleEncoding utf8
-  sydTest BaseSpec.spec
-  putStrLn "\n\n\n--------------------------------------------\n\n\n"
-  sydTest ExtraSpec.spec
 
-  putStrLn "\n\n\n--------------------------------------------\n\n\n"
+-- We use 'runPlutarchSpec' to test for unused goldens, but do so only in GHC
+-- 9. Because, under GHC 8 certain modules are disabled (see the CPP below)
```

Comment:
This assumption doesn't hold after #378

Reply:
in the sense that the same is true of GHC 9

Reply:
That PR hasn't changed anything in that regard. 

`Plutarch.FieldSpec` and `Plutarch.MonadicSpec` are still unused in GHC 8. In GHC 8 their associated goldens (see screenshot posted above) are seen incorrectly as unused goldens unless we do this CPP switch.

Reply:
> in the sense that the same is true of GHC 9

In practice, however, although we disable `Plutarch.FFISpec` in GHC 9 - that module is not generating any goldens, so it doesn't impact the unused goldens check. 

Would you instead go for a more general approach? We can use a known file path prefix for GHC 8 only goldens (Eg: `goldens/ghc8.field.data.bench.golden`) and automatically ignore them when checking under GHC 9.

Reply:
Honestly, I'd say don't bother. This is fine. It's not worth the effort.

Reply:
The proper solution would be to somehow accumulate state and run the process twice compiled with difference GHCs and then check that in CI.